### DISCUSSION TITLE
Best UI library for React projects in 2025

### DISCUSSION THREAD

**Discussion by koistya** [kriasoft/react-starter-kit#1977] [Ideas]
### ShadCN UI

![](https://img.shields.io/github/stars/shadcn-ui/ui?style=social)

- **Website**:  https://ui.shadcn.com/
- **GitHub**: https://github.com/shadcn-ui/ui

### Material UI, Joy UI

![](https://img.shields.io/github/stars/mui/material-ui?style=social)

- **Website**:  https://mui.com/
- **GitHub**: https://github.com/mui/material-ui

### Chakra UI

![](https://img.shields.io/github/stars/chakra-ui/chakra-ui?style=social)

- **Website**: https://chakra-ui.com/
- **GitHub**: https://github.com/chakra-ui/chakra-ui/

### Mantine

![](https://img.shields.io/github/stars/mantinedev/mantine?style=social)

- **Website**: https://mantine.dev/
- **GitHub**: https://github.com/mantinedev/mantine

### Radix UI

![](https://img.shields.io/github/stars/radix-ui/themes?style=social)

- **Website**: https://www.radix-ui.com/
- **GitHub**: https://github.com/radix-ui/themes

### PrimeReact

![](https://img.shields.io/github/stars/primefaces/primereact?style=social)

- **Website**: https://www.primefaces.org/primereact/
- **GitHub**: https://github.com/primefaces/primereact

---

### Discussions

- [`Reddit`] [What is your go-to UI library for production-grade react projects?](https://www.reddit.com/r/reactjs/comments/w8dn44/what_is_your_goto_ui_library_for_productiongrade/)

### Related Resources

- [The best state management library for a React project](https://github.com/kriasoft/react-starter-kit/discussions/1988)

**Comment by palaklive** [comment]
The innovative utilization of the server component approach surged interest in [radix-ui](https://www.radix-ui.com/). The harmonious fusion of [shadcn/ui](https://ui.shadcn.com/) with radix-ui and tailwind emerged as a standout combination.

**Reply by koistya** [reply]
Totally agree with this. The "it's not a component library" approach of shadcn/ui is such a breath of fresh air. Having the actual component code right in your project from the start seems like a perfect philosophy for a starter kit – maximum flexibility and no black boxes.
